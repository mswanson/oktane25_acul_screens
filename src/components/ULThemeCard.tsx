import { cn } from "@/lib/utils";

import { Card } from "./ui/card";

export interface ULThemeCardProps extends React.HTMLAttributes<HTMLDivElement> {
  /**
   * The content of the card element.
   */
  children: React.ReactNode;
  /**
   * Optional class names for additional styling or overriding default styles.
   */
  className?: string;
}

const ULThemeCard = ({ children, className, ...rest }: ULThemeCardProps) => {
  const themeClasses = cn(
    "px-10",
    "py-10",
    "theme-universal:bg-widget-bg",
    "theme-universal:border-(--color-widget-border)",
    "theme-universal:rounded-widget",
    "theme-universal:shadow-widget",
    "theme-universal:border-(length:--border-widget)",
    className
  );

  return (
    <Card className={themeClasses} {...rest}>
      {children}
    </Card>
  );
};

export default ULThemeCard;
