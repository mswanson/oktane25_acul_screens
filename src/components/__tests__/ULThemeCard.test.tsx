import { render } from "@testing-library/react";

import ULThemeCard from "../ULThemeCard";

describe("ULThemeCard Component", () => {
  // Snapshot Tests
  it("matches snapshot with default props", () => {
    const { container } = render(<ULThemeCard>Test Card</ULThemeCard>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with custom className", () => {
    const { container } = render(
      <ULThemeCard className="custom-class">Test Card</ULThemeCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with additional props", () => {
    const { container } = render(
      <ULThemeCard id="test-id" data-testid="card">
        Test Card
      </ULThemeCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with multiple children", () => {
    const { container } = render(
      <ULThemeCard>
        <span>Child 1</span>
        <span>Child 2</span>
      </ULThemeCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Functional Tests
  it("renders children correctly", () => {
    const { getByText } = render(<ULThemeCard>Test Card</ULThemeCard>);
    expect(getByText("Test Card")).toBeInTheDocument();
  });

  it("applies additional props correctly", () => {
    const { container } = render(
      <ULThemeCard data-testid="card">Test Card</ULThemeCard>
    );
    expect(container.firstChild).toHaveAttribute("data-testid", "card");
  });

  it("combines themed styles and custom className", () => {
    const { container } = render(
      <ULThemeCard className="custom-class">Test Card</ULThemeCard>
    );
    expect(container.firstChild).toHaveClass("custom-class");
    // Replace "themed-styles" with the actual default styles of ULThemeCard
    expect(container.firstChild).toHaveClass(
      "px-10 py-10 theme-universal:bg-widget-bg theme-universal:border-(--color-widget-border) theme-universal:rounded-widget theme-universal:shadow-widget theme-universal:border-(length:--border-widget)"
    );
  });
});
