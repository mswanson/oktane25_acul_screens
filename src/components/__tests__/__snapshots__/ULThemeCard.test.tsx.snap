// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeCard Component matches snapshot with additional props 1`] = `
<div
  class="bg-card text-card-foreground shadow-bevel-2xl flex flex-col gap-6 rounded-4xl px-10 py-10 theme-universal:bg-widget-bg theme-universal:border-(--color-widget-border) theme-universal:rounded-widget theme-universal:shadow-widget theme-universal:border-(length:--border-widget)"
  data-slot="card"
  data-testid="card"
  id="test-id"
>
  Test Card
</div>
`;

exports[`ULThemeCard Component matches snapshot with custom className 1`] = `
<div
  class="bg-card text-card-foreground shadow-bevel-2xl flex flex-col gap-6 rounded-4xl px-10 py-10 theme-universal:bg-widget-bg theme-universal:border-(--color-widget-border) theme-universal:rounded-widget theme-universal:shadow-widget theme-universal:border-(length:--border-widget) custom-class"
  data-slot="card"
>
  Test Card
</div>
`;

exports[`ULThemeCard Component matches snapshot with default props 1`] = `
<div
  class="bg-card text-card-foreground shadow-bevel-2xl flex flex-col gap-6 rounded-4xl px-10 py-10 theme-universal:bg-widget-bg theme-universal:border-(--color-widget-border) theme-universal:rounded-widget theme-universal:shadow-widget theme-universal:border-(length:--border-widget)"
  data-slot="card"
>
  Test Card
</div>
`;

exports[`ULThemeCard Component matches snapshot with multiple children 1`] = `
<div
  class="bg-card text-card-foreground shadow-bevel-2xl flex flex-col gap-6 rounded-4xl px-10 py-10 theme-universal:bg-widget-bg theme-universal:border-(--color-widget-border) theme-universal:rounded-widget theme-universal:shadow-widget theme-universal:border-(length:--border-widget)"
  data-slot="card"
>
  <span>
    Child 1
  </span>
  <span>
    Child 2
  </span>
</div>
`;
